openrouteservice/src/main/java/org/heigit/ors/matrix/algorithms/MatrixAlgorithmFactory.java

```diff
@@ -15,16 +15,21 @@
 
 import com.graphhopper.GraphHopper;
 import org.heigit.ors.matrix.MatrixRequest;
+import org.heigit.ors.matrix.algorithms.core.CoreMatrixAlgorithm;
 import org.heigit.ors.matrix.algorithms.dijkstra.DijkstraMatrixAlgorithm;
 import org.heigit.ors.matrix.algorithms.rphast.RPHASTMatrixAlgorithm;
+import org.heigit.ors.routing.graphhopper.extensions.ORSGraphHopper;
 
 public class MatrixAlgorithmFactory {
 	private MatrixAlgorithmFactory() {}
 
 	public static MatrixAlgorithm createAlgorithm(MatrixRequest req, GraphHopper gh) {
 		if (!req.getFlexibleMode() && gh.isCHEnabled())
 			return new RPHASTMatrixAlgorithm();
-		else
-			return new DijkstraMatrixAlgorithm();
+		if(gh instanceof ORSGraphHopper) {
+			if(req.getSearchParameters().getDynamicSpeeds() && ((ORSGraphHopper)gh).isCoreEnabled())
```

Comment:
I agree. This was added as a way to prevent people other than our specific customer from using it, as we have never discussed in ors whether we would like this to be the default matrix now. I do not see a reason for not using it, but I have added this "filter" so we can decide on it beforehand. In a way similar to the fast isochrones. What do you think?

Reply:
I believe `CoreMatrixAlgorithm` should be used in place of `DijkstraMatrixAlgorithm` for any matrix requests requiring additional edge filters.

Reply:
This is the same question as with Fast Isochrones. Let's add it to the code base first and then enable it after we have decided on it in the meeting.

Reply:
The only potential problem I can see with this approach is that by using the hidden `options` matrix API parameter one could now trigger the use of `DijkstraMatrixAlgorithm`. Such requests would use up considerably more hardware resources than RPHAST ones exposing a possible vulnerability of our service. @rabidllama Is this something we should be worried about?

Reply:
Plus even though `DijkstraMatrixAlgorithm` supports advanced route filters, it does not support turn restrictions 🙈

Reply:
What is the difference to before? Dijkstra was also available?

Reply:
As I see it the main difference lies in the fact that in practice it was not possible to trigger Dijkstra on our servers because with CH available RPHAST was always taking precedence.

The issue with turn restrictions is of course a different story. I've mentioned it here just in order to provide a more complete picture of the current situation.

Reply:
This is the `MatrixAlgorithmFactory`
```if (!req.getFlexibleMode() && gh.isCHEnabled())
			return new RPHASTMatrixAlgorithm();
		if(gh instanceof ORSGraphHopper) {
			if(req.getSearchParameters().getDynamicSpeeds() && ((ORSGraphHopper)gh).isCoreEnabled())
				return new CoreMatrixAlgorithm();
		}
		return new DijkstraMatrixAlgorithm();```

There was only a second if clause added. As you can see, CoreMatrix is only chosen with the `dynamicSpeeds` parameter. So I do not really understand what you mean. `DijkstraMatrixAlgorithm` is potentially called even less than before?

Reply:
My point is that it is possible now to force the public API to switch from RPHAST to Dijkstra which was not the case before.

Reply:
That was the case before as well with the `optimized` parameter.

Reply:
I can just remove the whole options parameter. Seeing how we "added" fast isochrones (i.e. we did not) to the service after almost 2 years, I do not see us having core matrix running anytime soon. So no use in having any options either.

Reply:
Oh, right, I missed that! Thanks for pointing it out.